src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharCategory.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.RegularExpressions
+{
+    internal class RegexCharCategory
+    {
+        private const byte Q = 5;    // quantifier
+        private const byte S = 4;    // ordinary stopper
+        private const byte Z = 3;    // ScanBlank stopper
+        private const byte X = 2;    // whitespace
+        private const byte E = 1;    // should be escaped
+
+        /// <summary>
+        /// For categorizing ASCII characters.
+        /// </summary>
+        private static readonly byte[] s_category = new byte[] {
```

Comment:
This should use `new ReadOnlySpan` to avoid the array allocation, like here: https://github.com/dotnet/coreclr/pull/20983/files#diff-7f94d87837f28201bc2b2b5887a49434R14

Reply:
thanks! does it need to have the hard coded array length for the mentioned code optimization to come in play?

Reply:
It does not need to - it can be just `[]`.